lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb

```diff
@@ -69,7 +69,7 @@ def self.invoke(obj, aem, args)
       aem.inputs.each do |f|
         key   = f.name
         value = args[key]
-        value = obj.attributes[key] || obj.substitute_value(f.default_value) if value.nil?
+        value = obj.attributes[key] || obj.get_value(f) if value.nil?
```

Comment:
@lfu does this field type have substitute set to true or nil?

Reply:
#<MiqAeField id: 48440, aetype: nil, name: "arg2", display_name: nil, datatype: "null coalescing", priority: 2, owner: nil, default_value: "${#var3} || ${#var2} || ${#var1} ", substitute: false, message: "create", visibility: nil, created_on: "2019-06-27 18:01:41", updated_on: "2019-08-20 19:22:05", class_id: nil, collect: nil, method_id: 11574, scope: nil, description: nil, condition: nil, on_entry: nil, on_exit: nil, on_error: nil, max_retries: nil, max_time: nil, updated_by: "admin", updated_by_user_id: 1>

Reply:
@mkanoor for null coalescing data type, substitute is false.

Reply:
@lfu 👍 
The NULL coalescing calls substitution directly